File: CsExample/EndpointDefinitions/SwaggerEndpointDefinition.cs
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.DependencyInjection;
using System.Threading.Tasks;

namespace CsExample.EndpointDefinitions
{
    public sealed class SwaggerEndpointDefinition : IEndpointDefinition
    {
        public void DefineServices(WebApplicationBuilder appBuilder)
        {
            appBuilder.Services
                .AddEndpointsApiExplorer()
                .AddSwaggerGen();
        }

        public void DefineEndpoints(WebApplication app)
        {
            app
                .UseSwagger()
                .UseSwaggerUI();

            app.MapGet("/", (HttpContext context) => {
                context.Response.Redirect("/swagger");
                return Task.CompletedTask;
            });
        }
    }
}

File: FsExample/EndpointDefinitions/SwaggerEndpointDefinition.fs
module SwaggerEndpointDefinition

open System.Threading.Tasks

open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Http
open Microsoft.Extensions.DependencyInjection

[<Sealed>]
type SwaggerEndpointDefinitions() =
    interface IEndpointDefinition with
        member _.DefineServices(appBuilder) =
            appBuilder.Services
                .AddEndpointsApiExplorer()
                .AddSwaggerGen()
            |> ignore

        member _.DefineEndpoints(app) =
            app.MapGet("/", fun (context: HttpContext) ->
                context.Response.Redirect("/swagger")
                Task.CompletedTask
                ) |> ignore

            app
                .UseSwagger()
                .UseSwaggerUI() |> ignore
